libraries/joomla/installer/adapters/component.php

```
@@ -1133,14 +1133,6 @@ public function uninstall($id)
 		$db->setQuery($query);
 		$db->execute();
 
-		// Check for errors.
-		if ($db->getErrorNum())
-		{
-			JLog::add(JText::_('JLIB_INSTALLER_ERROR_COMP_UNINSTALL_FAILED_DELETE_CATEGORIES'), JLog::WARNING, 'jerror');
```

Actually the installer will need some more thought, because you need to rollback changes. The installer is notorious for not giving good error messages so I think we'll need a different strategy.  I think the DB error message will be enough to pinpoint failure, but rather than logging the error, I think it would be better to JLog the successes so that a developer can trace through the point of failure.  So, every time we do something, we could do a:

`JLog::add(JText::_('JLIB_INSTALLER_LOG_SOMETHING'), JLog::INFO, 'jinstaller');`

com_installer can then add a message queue and possibly file log listener and then a big try-catch (where catch will rollback).

Does that make sense?